src/components/TaxonDetails/TaxonDetails.js

```diff
@@ -10,22 +11,24 @@ import {
 } from "react-native";
 import HTML from "react-native-render-html";
 
+import fetchTaxa from "../../lib/taxaFetchAPI";
 import { textStyles, viewStyles } from "../../styles/taxonDetails";
 import PhotoScroll from "../SharedComponents/PhotoScroll";
 import ViewWithFooter from "../SharedComponents/ViewWithFooter";
-import { useTaxonDetails } from "./hooks/useTaxonDetails";
 
 const TaxonDetails = ( ): React.Node => {
   const { params } = useRoute( );
   const { id } = params;
-  const { taxon, loading } = useTaxonDetails( id );
-  // const similarSpecies = useSimilarSpecies( id );
+  const {
+    data, isLoading, isError
+  } = useQuery( ["taxaFetch"], ( ) => fetchTaxa( id ) );
```

Comment:
I think here you want something like `useQuery( ["taxon", id], ( ) => fetchTaxa( id ) )` so the query key is specific to this taxon. If you don't, when you visit the next taxon detail view, the current taxon appears for a moment before the cache for that query key gets refreshed.